### PR TITLE
added subject alternate name support

### PR DESCRIPTION
**Pull request by schnipschnap**
Added support for SAN in certificates

### CODE REVIEW

File: core/mondoo-tls-security.mql.yaml [hunk 1]
```diff
@@ -72,11 +72,19 @@ queries:
...
+      checkA1 = tls.certificates.first.subject.commonName == asset.fqdn
+
       if(tls.certificates.first.subject.commonName.contains(/^\*/)) {
-        asset.fqdn.contains(tls.certificates.first.subject.commonName.split("*.")[1])
-      } else {
-        tls.certificates.first.subject.commonName == asset.fqdn
+        checkA1 = asset.fqdn.contains(tls.certificates.first.subject.commonName.split("*.")[1])
       }
+
+      checkA2 = tls.certificates.first.extensions.where(id=="2.5.29.17").first.value.split(" | ").contains(asset.fqdn)
```
Comment (chris-rock) [inline core/mondoo-tls-security.mql.yaml:81]: Why not using the sanExtension field?

```
cnspec shell host mondoo.com
→ connected to Network Host
  ___ _ __  ___ _ __   ___  ___ 
 / __| '_ \/ __| '_ \ / _ \/ __|
| (__| | | \__ \ |_) |  __/ (__ 
 \___|_| |_|___/ .__/ \___|\___|
   mondoo™     |_|              
cnspec> tls.certificates.first.sanExtension
tls.certificates.first.sanExtension: pkix.sanExtension dnsNames=[
  0: "mondoo.com"
  1: "assets.mondoo.com"
]
cnspec> tls.certificates { sanExtension } 
tls.certificates: [
  0: {
    sanExtension: pkix.sanExtension dnsNames=[
      0: "mondoo.com"
      1: "assets.mondoo.com"
    ]
  }
  1: {
    sanExtension: null
  }
  2: {
    sanExtension: null
  }
]
```

**Comment by schnipschnap** [comment]
@chris-rock good point thank you. Changed to your solution

File: core/mondoo-tls-security.mql.yaml [hunk 2]
```diff
@@ -72,11 +72,19 @@ queries:
...
-        asset.fqdn.contains(tls.certificates.first.subject.commonName.split("*.")[1])
-      } else {
-        tls.certificates.first.subject.commonName == asset.fqdn
+        checkA1 = asset.fqdn.contains(tls.certificates.first.subject.commonName.split("*.")[1])
       }
+
+      checkA2 = tls.certificates.first.sanExtension.dnsNames.contains(asset.fqdn)
+
+      wildC = "\*." + domainName.effectiveTLDPlusOne
+      checkA3 = wildC.in(tls.certificates.first.sanExtension.dnsNames)
```
Comment (mm-weber) [inline core/mondoo-tls-security.mql.yaml:84]:
```suggestion
      checkA3 = tls.certificates.first.sanExtension.dnsNames.any(_.split(".")[-2] + "." + _.split(".")[-1] == domainName.effectiveTLDPlusOne)
```

Comment (mm-weber) [reply core/mondoo-tls-security.mql.yaml:84]: This seems to work with any level of subdomain wildcards.

**Review by mm-weber — APPROVED** [review]
This might be more robust.

**Pull request merged**